<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="0" total="44" passed="44">
  <reporter-output>
  </reporter-output>
  <suite name="All Tests" duration-ms="9338" started-at="2015-11-23T09:58:34Z" finished-at="2015-11-23T09:58:44Z">
    <groups>
    </groups>
    <test name="Rest Integration Tests" duration-ms="8790" started-at="2015-11-23T09:58:34Z" finished-at="2015-11-23T09:58:43Z">
      <class name="rest_integration.PUT">
        <test-method status="PASS" signature="beforeClass()[pri:0, instance:rest_integration.PUT@12fad11]" name="beforeClass" is-config="true" duration-ms="4" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeClass -->
        <test-method status="PASS" signature="changeCategory()[pri:0, instance:rest_integration.PUT@12fad11]" name="changeCategory" duration-ms="663" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:42Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCategory -->
        <test-method status="PASS" signature="changeComment()[pri:0, instance:rest_integration.PUT@12fad11]" name="changeComment" duration-ms="347" started-at="2015-11-23T10:58:42Z" finished-at="2015-11-23T10:58:42Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeComment -->
        <test-method status="PASS" signature="changeItem()[pri:0, instance:rest_integration.PUT@12fad11]" name="changeItem" duration-ms="164" started-at="2015-11-23T10:58:42Z" finished-at="2015-11-23T10:58:43Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeItem -->
      </class> <!-- rest_integration.PUT -->
      <class name="rest_integration.GET">
        <test-method status="PASS" signature="beforeClass()[pri:0, instance:rest_integration.GET@19b4d0f]" name="beforeClass" is-config="true" duration-ms="195" started-at="2015-11-23T10:58:34Z" finished-at="2015-11-23T10:58:35Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeClass -->
        <test-method status="PASS" signature="allCommentsJSON()[pri:0, instance:rest_integration.GET@19b4d0f]" name="allCommentsJSON" duration-ms="6636" started-at="2015-11-23T10:58:35Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- allCommentsJSON -->
        <test-method status="PASS" signature="categoryHTML()[pri:0, instance:rest_integration.GET@19b4d0f]" name="categoryHTML" duration-ms="69" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- categoryHTML -->
        <test-method status="PASS" signature="categoryJSON()[pri:0, instance:rest_integration.GET@19b4d0f]" name="categoryJSON" duration-ms="19" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- categoryJSON -->
        <test-method status="PASS" signature="itemHTML()[pri:0, instance:rest_integration.GET@19b4d0f]" name="itemHTML" duration-ms="31" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- itemHTML -->
        <test-method status="PASS" signature="commentHTML()[pri:0, instance:rest_integration.GET@19b4d0f]" name="commentHTML" duration-ms="13" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- commentHTML -->
        <test-method status="PASS" signature="itemJSON()[pri:0, instance:rest_integration.GET@19b4d0f]" name="itemJSON" duration-ms="11" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- itemJSON -->
        <test-method status="PASS" signature="commentJSON()[pri:0, instance:rest_integration.GET@19b4d0f]" name="commentJSON" duration-ms="13" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- commentJSON -->
        <test-method status="PASS" signature="allCommentsHTML()[pri:0, instance:rest_integration.GET@19b4d0f]" name="allCommentsHTML" duration-ms="18" started-at="2015-11-23T10:58:41Z" finished-at="2015-11-23T10:58:41Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- allCommentsHTML -->
      </class> <!-- rest_integration.GET -->
      <class name="rest_integration.POST">
        <test-method status="PASS" signature="beforeClass()[pri:0, instance:rest_integration.POST@1698539]" name="beforeClass" is-config="true" duration-ms="1" started-at="2015-11-23T10:58:43Z" finished-at="2015-11-23T10:58:43Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeClass -->
        <test-method status="PASS" signature="newCategory()[pri:0, instance:rest_integration.POST@1698539]" name="newCategory" duration-ms="230" started-at="2015-11-23T10:58:43Z" finished-at="2015-11-23T10:58:43Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newCategory -->
        <test-method status="PASS" signature="newComment()[pri:0, instance:rest_integration.POST@1698539]" name="newComment" duration-ms="150" started-at="2015-11-23T10:58:43Z" finished-at="2015-11-23T10:58:43Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newComment -->
        <test-method status="PASS" signature="newItem()[pri:0, instance:rest_integration.POST@1698539]" name="newItem" duration-ms="165" started-at="2015-11-23T10:58:43Z" depends-on-methods="rest_integration.POST.newCategory" finished-at="2015-11-23T10:58:43Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newItem -->
      </class> <!-- rest_integration.POST -->
    </test> <!-- Rest Integration Tests -->
    <test name="Rest Unit Tests" duration-ms="519" started-at="2015-11-23T09:58:43Z" finished-at="2015-11-23T09:58:44Z">
      <class name="rest_unit.CommentControllerTest">
        <test-method status="PASS" signature="beforeClass()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="beforeClass" is-config="true" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeClass -->
        <test-method status="PASS" signature="changeComment()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="changeComment" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeComment -->
        <test-method status="PASS" signature="changeCommentNoPermission()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="changeCommentNoPermission" duration-ms="0" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCommentNoPermission -->
        <test-method status="PASS" signature="changeCommentNotInTheDatabase()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="changeCommentNotInTheDatabase" duration-ms="0" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.CommentController.changeComment(CommentController.java:45)
	at rest_unit.CommentControllerTest.changeCommentNotInTheDatabase(CommentControllerTest.java:77)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCommentNotInTheDatabase -->
        <test-method status="PASS" signature="changeCommentOwnComment()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="changeCommentOwnComment" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCommentOwnComment -->
        <test-method status="PASS" signature="deleteComment()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="deleteComment" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteComment -->
        <test-method status="PASS" signature="deleteCommentNoPermission()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="deleteCommentNoPermission" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteCommentNoPermission -->
        <test-method status="PASS" signature="deleteCommentNotInTheDatabase()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="deleteCommentNotInTheDatabase" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.CommentController.deleteComment(CommentController.java:60)
	at rest_unit.CommentControllerTest.deleteCommentNotInTheDatabase(CommentControllerTest.java:96)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteCommentNotInTheDatabase -->
        <test-method status="PASS" signature="newComment()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="newComment" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newComment -->
        <test-method status="PASS" signature="newCommentNoPermission()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="newCommentNoPermission" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newCommentNoPermission -->
        <test-method status="PASS" signature="newCommentNoText()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="newCommentNoText" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newCommentNoText -->
        <test-method status="PASS" signature="newCommentNotInTheDatabase()[pri:0, instance:rest_unit.CommentControllerTest@167135c]" name="newCommentNotInTheDatabase" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.CommentController.newComment(CommentController.java:74)
	at rest_unit.CommentControllerTest.newCommentNotInTheDatabase(CommentControllerTest.java:122)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newCommentNotInTheDatabase -->
      </class> <!-- rest_unit.CommentControllerTest -->
      <class name="rest_unit.ItemControllerTest">
        <test-method status="PASS" signature="beforeClass()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="beforeClass" is-config="true" duration-ms="10" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeClass -->
        <test-method status="PASS" signature="changeItem()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="changeItem" duration-ms="5" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeItem -->
        <test-method status="PASS" signature="changeItemNoPermission()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="changeItemNoPermission" duration-ms="5" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeItemNoPermission -->
        <test-method status="PASS" signature="changeItemNotInTheDatabase()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="changeItemNotInTheDatabase" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.ItemController.changeItem(ItemController.java:44)
	at rest_unit.ItemControllerTest.changeItemNotInTheDatabase(ItemControllerTest.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeItemNotInTheDatabase -->
        <test-method status="PASS" signature="changeItemWithoutTitleDescription()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="changeItemWithoutTitleDescription" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeItemWithoutTitleDescription -->
        <test-method status="PASS" signature="deleteItem()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="deleteItem" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteItem -->
        <test-method status="PASS" signature="deleteItemNoPermission()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="deleteItemNoPermission" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteItemNoPermission -->
        <test-method status="PASS" signature="deleteItemNotInTheDatabase()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="deleteItemNotInTheDatabase" duration-ms="0" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.ItemController.deleteItem(ItemController.java:60)
	at rest_unit.ItemControllerTest.deleteItemNotInTheDatabase(ItemControllerTest.java:110)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteItemNotInTheDatabase -->
        <test-method status="PASS" signature="newItem()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="newItem" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newItem -->
        <test-method status="PASS" signature="newItemItemNoPermission()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="newItemItemNoPermission" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newItemItemNoPermission -->
        <test-method status="PASS" signature="newItemItemWithoutTitle()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="newItemItemWithoutTitle" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newItemItemWithoutTitle -->
        <test-method status="PASS" signature="newItemNoDescription()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="newItemNoDescription" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newItemNoDescription -->
        <test-method status="PASS" signature="newItemNotInTheDatabase()[pri:0, instance:rest_unit.ItemControllerTest@1e46e8f]" name="newItemNotInTheDatabase" duration-ms="0" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.ItemController.newItem(ItemController.java:73)
	at rest_unit.ItemControllerTest.newItemNotInTheDatabase(ItemControllerTest.java:143)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- newItemNotInTheDatabase -->
      </class> <!-- rest_unit.ItemControllerTest -->
      <class name="rest_unit.CategoryControllerTest">
        <test-method status="PASS" signature="beforeClass()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="beforeClass" is-config="true" duration-ms="381" started-at="2015-11-23T10:58:43Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeClass -->
        <test-method status="PASS" signature="changeCategory()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="changeCategory" duration-ms="6" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCategory -->
        <test-method status="PASS" signature="changeCategoryNoPermission()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="changeCategoryNoPermission" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCategoryNoPermission -->
        <test-method status="PASS" signature="changeCategoryWrongCategory()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="changeCategoryWrongCategory" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCategoryWrongCategory -->
        <test-method status="PASS" signature="changeCategoryWrongUser()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="changeCategoryWrongUser" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.CategoryController.changeCategory(CategoryController.java:36)
	at rest_unit.CategoryControllerTest.changeCategoryWrongUser(CategoryControllerTest.java:58)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- changeCategoryWrongUser -->
        <test-method status="PASS" signature="createCategory()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="createCategory" duration-ms="5" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- createCategory -->
        <test-method status="PASS" signature="createCategoryNoPermission()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="createCategoryNoPermission" duration-ms="2" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- createCategoryNoPermission -->
        <test-method status="PASS" signature="createCategoryWrongUser()[pri:0, instance:rest_unit.CategoryControllerTest@837a77]" name="createCategoryWrongUser" duration-ms="1" started-at="2015-11-23T10:58:44Z" finished-at="2015-11-23T10:58:44Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at rest.controller.CategoryController.newCategory(CategoryController.java:25)
	at rest_unit.CategoryControllerTest.createCategoryWrongUser(CategoryControllerTest.java:77)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1215)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.run(TestNG.java:1048)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:112)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:205)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:176)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- createCategoryWrongUser -->
      </class> <!-- rest_unit.CategoryControllerTest -->
    </test> <!-- Rest Unit Tests -->
  </suite> <!-- All Tests -->
</testng-results>
